### ISSUE TITLE
Use skopeo to pull container images

### ISSUE THREAD

**Issue by rnjudge** [tern-tools/tern#948]
**Background**
Reference: https://kubernetes.io/blog/2020/12/02/dont-panic-kubernetes-and-docker/ 

TL;DR - Kubernetes currently uses Dockershim to access containerd container runtime. Dockershim is being removed from Kubelet which removes support for Docker as a container runtime in favor of other runtimes that use [Container Runtime Interface](https://kubernetes.io/blog/2016/12/container-runtime-interface-cri-in-kubernetes/) created for K8s.

We would like Tern to run in Kubernetes. This means that we cannot use Docker commands to pull images from the registry with the`-i` command line option if we want Tern to run in Kubernetes after v1.2.0. 

**Proposal**
Let's try to use `subprocess` + [ORAS](https://github.com/deislabs/oras) instead of the Docker python library to pull container images from a registry when running with the -i option. This will mean that we need to include the go module as a dependency of Tern. 
* Create an OCI image class and implement the `load_image(`) method for it in accordance with how ORAS pulls images.
* We may need an additional tool on top of ORAS. ORAS will send all the bytes of the downloaded image to stdout and we need to figure out how we store these results to analyze.

**Use Cases**
* Deploying Tern as a k8s cluster on a CRD

**Comment by rnjudge** [comment]
It does not appear that ORAS supports anonymous pulling of container images from Dockerhub at this time. Skopeo does support this behavior and therefore, we should explore using skopeo instead. A previous PR (https://github.com/tern-tools/tern/pull/712) started to work on this and we may want to revive parts of this.

**Comment by falcorocks** [comment]
+1 on skopeo, I have used it to successfully achieve something similar to what you are trying to do here.
Abandoning docker in favour of something like skopeo would also make it much better for CI, where docker in docker is still a nogo for security reasons!

**Comment by rnjudge** [comment]
@mukultaneja is working on this. @falcorocks if you have any tips or pointers, feel free to drop them here :)

**Comment by mukultaneja** [comment]
@rnjudge @nishakm, how are we going to make `skopeo` as a requirement to tern moving onwards?
